cpp/src/io/parquet/decode_fixed.cu

```diff
@@ -560,6 +820,22 @@ __device__ inline bool maybe_has_nulls(page_state_s* s)
   return run_val != s->col.max_level[lvl];
 }
 
+template <int decode_block_size_t, typename stream_type>
+__device__ int skip_decode(stream_type& parquet_stream, int num_to_skip, int t)
+{
+  // it could be that (e.g.) we skip 5000 but starting at row 4000 we have a run of length 2000:
+  // in that case skip_decode() only skips 4000, and we have to process the remaining 1000 up front
+  // modulo 2 * block_size of course, since that's as many as we process at once
+  int num_skipped = parquet_stream.skip_decode(t, num_to_skip);
+  while (num_skipped < num_to_skip) {
+    auto const to_decode = min(2 * decode_block_size_t, num_to_skip - num_skipped);
+    num_skipped += parquet_stream.decode_next(t, to_decode);
```

Comment:
Future optimization:  add functionality so that decode_next() doesn't actually perform the decode.  All we really need is for it to march the run headers and skip to the appropriate location.

Reply:
Added a comment to this effect.